test/geometry/test_reshape.py

```diff
@@ -1172,3 +1174,61 @@ def test_wraparound_closed_linestring_partially_reshaped(
     )
 
     _assert_layer_geoms(layer, [result])
+
+
+@pytest.mark.parametrize(
+    argnames=(
+        "new_start_wkt",
+        "new_end_wkt",
+        "expected_update_geom_count",
+    ),
+    argvalues=[
+        (
+            "POINT(0 0)",
+            "POINT(0 2)",
+            0,
+        ),
+        (
+            "POINT(1 0)",
+            "POINT(0 2)",
+            1,
+        ),
+    ],
+    ids=["edge not moved", "edge moved"],
+)
+def test_move_edges(
+    preset_features_layer_factory: Callable[
+        [str, list[str]], tuple[QgsVectorLayer, list[QgsFeature]]
+    ],
+    new_start_wkt: str,
+    new_end_wkt: str,
+    mocker: MockerFixture,
+    expected_update_geom_count: int,
+):
+    m_set_editable_and_begin_edit_command_once = mocker.patch.object(
+        reshape, "_set_editable_and_begin_edit_command_once", autospec=True
+    )
+    m_update_geometry_to_layer_feature = mocker.patch.object(
+        reshape, "_update_geometry_to_layer_feature", autospec=True
+    )
+
+    new_start = QgsPoint()
+    assert new_start.fromWkt(new_start_wkt)
+
+    new_end = QgsPoint()
+    assert new_end.fromWkt(new_end_wkt)
+
+    layer1, features1 = preset_features_layer_factory(
+        "l1", ["POINT(0 0)", "POINT(2 2)"]
+    )
+    edges = [
+        ReshapeEdge(layer1, features1[0], 0, is_start=True),
+    ]
+
+    _move_edges(edges, new_start, new_end)
+
+    assert (
+        m_set_editable_and_begin_edit_command_once.call_count
```

Comment:
Could be more robust to test for layers being editable after the operation, instead of mocking these calls?

Reply:
Yes that is a better option, changed the test